src/filesystem.cc

```diff
@@ -1941,39 +2110,283 @@ GetFileSystem(const std::string& path, FileSystem** file_system)
+    std::string file_path = std::string(file_path_c_str);
+    LOG_VERBOSE(1) << "Reading cloud credential from " << file_path;
+
+    triton::common::TritonJson::Value creds_json;
+    std::string cred_file_content;
+    RETURN_IF_ERROR(local_fs_->ReadTextFile(file_path, &cred_file_content));
+    RETURN_IF_ERROR(creds_json.Parse(cred_file_content));
+
+#ifdef TRITON_ENABLE_GCS
+    // load GCS credentials
+    gs_cache_.clear();
+    triton::common::TritonJson::Value gs_creds_json;
+    if (creds_json.Find("gs", &gs_creds_json)) {
+      std::vector<std::string> gs_cred_names;
+      gs_creds_json.Members(&gs_cred_names);
+      for (size_t i = 0; i < gs_cred_names.size(); i++) {
+        std::string gs_cred_name = gs_cred_names[i];
+        std::string gs_cred_path;
+        triton::common::TritonJson::Value gs_cred_path_json;
+        gs_creds_json.Find(gs_cred_name.c_str(), &gs_cred_path_json);
+        gs_cred_path_json.AsString(&gs_cred_path);
+        gs_cache_.push_back(std::make_tuple(
+            gs_cred_name, gs_cred_path, std::shared_ptr<GCSFileSystem>()));
+      }
+      SortCache(gs_cache_);
+    }
+#endif  // TRITON_ENABLE_GCS
+
+#ifdef TRITON_ENABLE_S3
+    // load S3 credentials
+    s3_cache_.clear();
+    triton::common::TritonJson::Value s3_creds_json;
+    if (creds_json.Find("s3", &s3_creds_json)) {
+      std::vector<std::string> s3_cred_names;
+      s3_creds_json.Members(&s3_cred_names);
+      for (size_t i = 0; i < s3_cred_names.size(); i++) {
+        std::string s3_cred_name = s3_cred_names[i];
+        std::string s3_cred_secret_key, s3_cred_key_id, s3_cred_region,
+            s3_cred_session_token, s3_cred_profile;
+        triton::common::TritonJson::Value s3_cred_json;
+        s3_creds_json.Find(s3_cred_name.c_str(), &s3_cred_json);
+        triton::common::TritonJson::Value s3_cred_secret_key_json,
+            s3_cred_key_id_json, s3_cred_region_json,
+            s3_cred_session_token_json, s3_cred_profile_json;
+        if (s3_cred_json.Find("secret_key", &s3_cred_secret_key_json))
+          s3_cred_secret_key_json.AsString(&s3_cred_secret_key);
+        if (s3_cred_json.Find("key_id", &s3_cred_key_id_json))
+          s3_cred_key_id_json.AsString(&s3_cred_key_id);
+        if (s3_cred_json.Find("region", &s3_cred_region_json))
+          s3_cred_region_json.AsString(&s3_cred_region);
+        if (s3_cred_json.Find("session_token", &s3_cred_session_token_json))
+          s3_cred_session_token_json.AsString(&s3_cred_session_token);
+        if (s3_cred_json.Find("profile", &s3_cred_profile_json))
+          s3_cred_profile_json.AsString(&s3_cred_profile);
+        s3_cache_.push_back(std::make_tuple(
+            s3_cred_name,
+            std::make_tuple(
+                s3_cred_secret_key, s3_cred_key_id, s3_cred_region,
+                s3_cred_session_token, s3_cred_profile),
+            std::shared_ptr<S3FileSystem>()));
+      }
+      SortCache(s3_cache_);
+    }
```

Comment:
Can these be moved into internal helpers like `LoadGCSCredentials`, `LoadS3Crendentials`, etc. for brevity?

Reply:
Yes. Updated.